apimonitor/views.py

```diff
@@ -4,20 +4,119 @@
 from django.db.models import Avg, Count, Q
 from django.utils import timezone
 from django.conf import settings
-from rest_framework import viewsets, mixins
+from rest_framework import viewsets, mixins, status
 from rest_framework.response import Response
 from rest_framework.permissions import IsAuthenticated
 
-from apimonitor.models import APIMonitor, APIMonitorResult
-from apimonitor.serializers import APIMonitorSerializer, APIMonitorListSerializer
+from apimonitor.models import (APIMonitor, APIMonitorResult, APIMonitorQueryParam,
+                               APIMonitorHeader, APIMonitorBodyForm, APIMonitorRawBody)
+from apimonitor.serializers import (APIMonitorSerializer, APIMonitorListSerializer,
+                                    APIMonitorQueryParamSerializer, APIMonitorHeaderSerializer,
+                                    APIMonitorBodyFormSerializer, APIMonitorRawBodySerializer)
 
 
-class APIMonitorViewSet(mixins.ListModelMixin, mixins.DestroyModelMixin, viewsets.GenericViewSet):
+class APIMonitorViewSet(mixins.ListModelMixin,
+                        mixins.CreateModelMixin,
+                        mixins.DestroyModelMixin,
+                        viewsets.GenericViewSet):
+
     queryset = APIMonitor.objects.all()
     serializer_class = APIMonitorSerializer
     permission_classes = [IsAuthenticated]
     lookup_field = "pk"
 
+    def create(self, request, *args, **kwargs):
+        monitor_data = {
+            'user': request.user,
+            'name': request.data.get('name'),
+            'method': request.data.get('method'),
+            'url': request.data.get('url'),
+            'schedule': request.data.get('schedule'),
+            'body_type': request.data.get('body_type')
+        }
+        api_monitor_serializer = APIMonitorSerializer(data=monitor_data)
+        if (api_monitor_serializer.is_valid()):
+            monitor_obj = APIMonitor.objects.create(**monitor_data)
+            error_log = []
+            try:
+                if (request.data.get('query_params')):
+                    for key_value_pair in request.data.get('query_params'):
+                        if (len(key_value_pair) == 2):
+                            key, value = key_value_pair[0], key_value_pair[1]
+                        else:
+                            error_log += ["Please make sure your [query params] is in the form of [[key1, value1], [key2,value2], ...]"]
```

Comment:
```suggestion
                            error_log += ["Please make sure you submit correct query params"]
```
ini kayaknya gaperlu specific karena user gak tau juga wkkwk